keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -307,22 +313,41 @@ def __init__(
                 )
 
         super().__init__(**kwargs)
+        
+        if vocabulary_size is not None:
+            if isinstance(vocabulary_size, int) == False:
+                raise ValueError(
+                    "vocabulary size must be int. "
+                    f"Recieved: type {type(vocabulary_size)}"
+                )
 
         if isinstance(vocabulary, str):
             self.vocabulary = [
                 line.rstrip() for line in tf.io.gfile.GFile(vocabulary)
             ]
+            if vocabulary_size is None:
+                vocabulary_size = len(self.vocabulary)
+            if len(self.vocabulary) < vocabulary_size:
+                warnings.warn("Setting vocab size to a larger value than the input vocabulary file.\
+￼                        Some token ids will never be output from the tokenizer.")
+                vocabulary_size = len(self.vocabulary)
+            else:
+                self.vocabulary = self.vocabulary[0:vocabulary_size]
         elif isinstance(vocabulary, Iterable):
             # Make a copy.
             self.vocabulary = list(vocabulary)
+            if vocabulary_size is None:
+                vocabulary_size = len(self.vocabulary)
+            else:
+                self.vocabulary = self.vocabulary[0:vocabulary_size]
         else:
             raise ValueError(
                 "Vocabulary must be an file path or list of terms. "
                 f"Received: vocabulary={vocabulary}"
             )
         if oov_token is None:
             raise ValueError("`oov_token` cannot be None.")
-
+        
```

Comment:
Please format the script using `shell/format.sh`. This will remove all these extra spaces. And also, please avoid making any changes unrelated to the PR.

Reply:
ok i am looking into it